collections/src/test/kotlin/ktx/collections/arraysTest.kt

```diff
@@ -364,6 +364,25 @@ class ArraysTest {
     assertEquals(GdxArray.with("Twenty-one", "Eleven", "One"), array)
   }
 
+  @Test
+  fun `should remove elements from existing GdxArray`() {
+    val array = GdxArray.with(1, 2, 3, 4, 5)
+
+    array.removeAll { it % 2 == 0 }
+
+    assertEquals(GdxArray.with(1, 3, 5), array)
+  }
+
+  @Test
+  fun `should retain elements from existing GdxArray`() {
+    val array = GdxArray.with(1, 2, 3, 4, 5)
+
+    array.retainAll { it % 2 == 1 }
+
+    assertEquals(GdxArray.with(1, 3, 5), array)
+  }
+
```

Comment:
Add tests for edge cases: empty arrays, no elements retained/removed.

Reply:
Also added cases for all elements retained/removed.